network-area-diagram/src/main/java/com/powsybl/nad/layout/ImprovedForceLayoutFactory.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.nad.layout;
+
+import com.powsybl.nad.model.Point;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * @author Sophie Frasnedo {@literal <sophie.frasnedo at rte-france.com>}
+ */
+public class ImprovedForceLayoutFactory implements LayoutFactory {
+
+    private final Map<String, Point> fixedPositions;
+
+    public ImprovedForceLayoutFactory(Map<String, Point> fixedPositions) {
+        Objects.requireNonNull(fixedPositions);
+        this.fixedPositions = fixedPositions;
+    }
+
+    @Override
+    public Layout create() {
+        AbstractLayout layout = new BasicForceLayout();
```

Comment:
Here's `BasicForceLayout` but in fact we may want fixed positions on any `Layout` algorithms, so it would be nicer to have a factory as constructor argument.

If that factory is null, the fallback could be the `BasicFixedLayout`